web/src/js/ducks/modes/wireguard.ts

```diff
@@ -21,21 +22,23 @@ export const initialState: WireguardState = {
 };
 
 export const getMode = (modes) => {
-    const wireguardMode = modes.wireguard;
+    const wireguardMode: WireguardState = modes.wireguard;
     return includeModeState("wireguard", wireguardMode);
 };
 
-export const toggleWireguard = () => {
-    return async (dispatch) => {
+export const toggleWireguard =
+    (updateModeFunc = updateMode) =>
```

Comment:
Is there a reason why we don't mock the fetch itself? :)

Reply:
what do you mean?

Reply:
Make toggleWireguard not accept any arguments, and in tests we mock the fetch instead of updateMode.

Reply:
it's so annoying that with `fetchMock` I cannot simulate properly the error and have the `state.error`updated